### PR TITLE
Add codecov for unit test coverage

### PR DESCRIPTION
**Pull request by varshaprasad96**
# Issue link

# What changes have been made

# Verification steps

## Checks
- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

**Comment by varshaprasad96** [comment]
@ChristianZaccaria Could we merge this to check if coverage reports are being uploaded? Looks like it should automatically search for coverage files in the repo and add it - so no more path configuration is needed.

**Review by Ygnas — APPROVED** [review]

**Review by ChristianZaccaria — APPROVED** [review]
Good for merge on my side
/lgtm
/approve

File: .github/workflows/unit-tests.yml
Comment (ChristianZaccaria) [inline .github/workflows/unit-tests.yml:31]: On second thought, I think the code coverage report won't be uploaded if the previous step results in less than 90% coverage, exiting the workflow.

**Review by ChristianZaccaria — CHANGES_REQUESTED** [review]
One thought on this actually:

File: .github/workflows/unit-tests.yml
Comment (varshaprasad96) [reply .github/workflows/unit-tests.yml:31]: That's a good catch! It should work now.

**Comment by varshaprasad96** [comment]
/hold

The action seems to be failing with rate limiting error mentioning that a new token is required.

**Review by KPostOffice — APPROVED** [review]

**Review by ChristianZaccaria — APPROVED** [review]
/lgtm

**Pull request merged**
